ui/components/grant-form.js

```diff
@@ -0,0 +1,133 @@
+import { useEffect, useState, useRef } from 'react'
+import useSWR, { useSWRConfig } from 'swr'
+import { Combobox } from '@headlessui/react'
+import { CheckIcon } from '@heroicons/react/solid'
+import { PlusIcon } from '@heroicons/react/outline'
+
+import RoleDropdown from './role-dropdown'
+
+export default function ({ resource, roles }) {
```

Comment:
Nit: this component should named rather than an unnamed function.

Ex: `export default function GrantForm`